lib/ansible/module_utils/junos.py

```diff
@@ -40,7 +40,7 @@
     'password': dict(fallback=(env_fallback, ['ANSIBLE_NET_PASSWORD']), no_log=True),
     'ssh_keyfile': dict(fallback=(env_fallback, ['ANSIBLE_NET_SSH_KEYFILE']), type='path'),
     'timeout': dict(type='int', default=10),
-    'provider': dict(type='dict'),
+    'provider': dict(type='dict', no_log=True),
```

Comment:
I think this is a regression in the argspec work that went into 2.3 for Network modules, raised as https://github.com/ansible/ansible/issues/21892